packages/playwright-core/src/client/elementHandle.ts

```diff
@@ -262,18 +262,37 @@ export async function convertInputFiles(files: string | FilePayload | string[] |
   if (items.some(item => typeof item === 'string')) {
     if (!items.every(item => typeof item === 'string'))
       throw new Error('File paths cannot be mixed with buffers');
+    const itemFileTypes = (await Promise.all((items as string[]).map(async item => (await fs.promises.stat(item)).isDirectory() ? 'directory' : 'file')));
```

Comment:
I think it would be easier to split `items` into `files` and `directory` variables here, and not use `items` in the code below. WDYT?